packages/sso-menu/src/SSOMenu.tsx

```diff
@@ -0,0 +1,250 @@
+
+  /**
+   * MongoDB product that is currently active: ['atlas', 'university', 'support'].
+   */
+  activeProduct: ActiveProduct;
+
+  /**
+   * Callback invoked when user logs out.
+   */
+  onLogout?: React.MouseEventHandler;
+
+  /**
+   * Callback invoked when user switches products.
+   */
+  onProductChange?: React.MouseEventHandler;
+
+  /**
+   * Callback invoked when user views their MongoDB account.
+   */
+  onAccountClick?: React.MouseEventHandler;
+}
+
+/**
+ * # SSOMenu
+ *
+ * ```
+<SSOMenu
+    user={{ name: 'Alex Smith', email: 'alex.smith@youwork.com' }}
+    activeProduct="atlas"
+    onAccountClick={() => console.log('Viewing account information')}
+    onLogout={() => console.log('On logout')}
+    onProductChange={() => console.log('Switching products')}
+  />
+ * ```
+ * @param props.user Object that contains information about the active user. {name: 'string', email: 'string'}
+ * @param props.activeProduct  MongoDB product that is currently active: ['atlas', 'university', 'support'].
+ * @param props.onLogout Callback invoked when user logs out.
+ * @param props.onAccountClick Callback invoked when user views their MongoDB account.
+ * @param props.onProductChange Callback invoked when user switches products.
+ *
+ */
+function SSOMenu({
+  user: { name, email },
+  activeProduct,
+  onLogout,
+  onProductChange,
+  onAccountClick,
+}: SSOMenuProps) {
+  const triggerRef = useRef<HTMLButtonElement>(null);
+
+  return (
+    <Menu
+      trigger={
+        <button ref={triggerRef} type="button" className={menuButtonStyle}>
+          <span className={menuNameStyle}>{name}</span>
+          <Icon glyph="CaretDown" className={iconStyle} />
+        </button>
+      }
+    >
+      <MenuGroup className={accountMenuGroupStyle}>
+        <h3 className={cx(nameStyle, truncate)}>{name}</h3>
+        <p className={descriptionStyle}>{email}</p>
+        <Button
+          size="small"
+          onClick={onAccountClick}
+          className={accountButtonStyle}
+          as="a"
+        >
+          MongoDB Account
+        </Button>
+      </MenuGroup>
+      <MenuGroup>
+        {menuItems.map(el => (
+          <MenuItem
+            onClick={onProductChange}
+            key={el.displayName}
+            active={el.slug === activeProduct}
+            className={menuItemPadding}
+            href={el.href}
+          >
+            <p className={menuItemTextStyle}>{el.displayName}</p>
+            <p className={descriptionStyle}>{el.description}</p>
+          </MenuItem>
+        ))}
+      </MenuGroup>
+      <MenuItem
+        active={false}
+        onClick={onLogout}
+        className={cx(
+          logoutContainerHeight,
+          menuItemTextStyle,
+          menuItemPadding,
+        )}
+      >
+        Logout
+      </MenuItem>
+    </Menu>
+  );
+}
+
+SSOMenu.displayName = 'SSOMenu';
+
+SSOMenu.propTypes = {
+  user: PropTypes.object,
+  activeProduct: PropTypes.oneOf(['atlas', 'support', 'university']),
```

Comment:
We should generate this from `menuItems`.

Reply:
Sounds good!